### ISSUE TITLE
Interesting behaviour when migrations run 

### ISSUE THREAD

**Issue by Junicus** [2020IP/TwentyTwenty.IdentityServer3.EntityFramework7#5]
I'm not exactly sure why it's happening but when I register the dbcontexts with different connection strings I always get the same connection string.  

In ConfigureServices:

```
services.AddEntityFramework()
.AddSqlServer()
.AddDbContext<OperationalContext>(o => o.UseSqlServer(Configuration["IdentityServer3:ConnectionStrings:Identity_Operational"]))
.AddDbContext<ClientConfigurationContext>(o => o.UseSqlServer(Configuration["IdentityServer3:ConnectionStrings:Identity_Clients"]))
.AddDbContext<ScopeConfigurationContext>(o => o.UseSqlServer(Configuration["IdentityServer3:ConnectionStrings:Identity_Scopes"]));
```

In Configure:

```
var factory = new IdentityServerServiceFactory();
factory.UseInMemoryUsers(Users.Get())
 .ConfigureEntityFramework(app.ApplicationServices)
  .RegisterOperationalStores<OperationalContext>()
  .RegisterClientStore<Guid, ClientConfigurationContext>()
  .RegisterScopeStore<Guid, ScopeConfigurationContext>();
```

The contexts:

```
    public class OperationalContext : IDEF.OperationalContext
    {
        public OperationalContext(DbContextOptions options) : base(options)
        {
            Database.EnsureCreated();
        }
    }

    public class ClientConfigurationContext : IDEF.ClientConfigurationContext<Guid>
    {
        public ClientConfigurationContext(DbContextOptions options) : base(options)
        {
            Database.EnsureCreated();
        }
    }

    public class ScopeConfigurationContext : IDEF.ScopeConfigurationContext<Guid>
    {
        public ScopeConfigurationContext(DbContextOptions options) : base(options)
        {
            Database.EnsureCreated();
        }
    }
```

And finally in appsettings.json:

```
{
  "IdentityServer3": {
    "ConnectionStrings": {
      "Identity_Clients": "Server=.\\sqlexpress;Database=Identity_Clients;Trusted_Connection=True;",
      "Identity_Scopes": "Server=.\\sqlexpress;Database=Identity_Scopes;Trusted_Connection=True;",
      "Identity_Operational": "Server=.\\sqlexpress;Database=Identity_Operational;Trusted_Connection=True;",
      "Identity_Users": "Server=.\\sqlexpress;Database=Identity_Users;Trusted_Connection=True;"
    }
  }
}
```

When running the site, it says that Clients is an invalid object, and it is, the database has not been created.  Only the scopes database is created.  I placed a few breakpoints and the options getting to all 3 contexts are the same, they are Identity_Scopes, guessing because it's the last one registered.  Any ideas?  I'm in the mind of scripting the database with all the tables in the same one and just using 1 connection string, but wanted to see more clearly how it all ties in during development.

It might also be a EF7 thing

**Comment by ericgreenmix** [comment]
Could you post a link to a repo that reproduces this issue? Having a hard time determining what the possible cause could be from the snippets.

**Comment by Junicus** [comment]
Will work on a sample repo with the problem for you, I will post the link as soon as I have it.

**Comment by Junicus** [comment]
https://github.com/Junicus/Sample2020

Again, this only happens when you are trying to do different databases.  In the case of this example, I want the scopes, clients, and operational tables in different databases.

Yes it's silly and I ended up making a single database for the whole identity story.  I get that normally the contexts in the sample would be in the same database, but if you were to do a database for client, scope, operational and a database for users, you run into this issue.

I also made some debug commands on both projects so the solution can be set to multiple sites.  Let me know if that helps you.
